processing/src/main/java/org/apache/druid/segment/IndexMergerV9.java

```diff
@@ -332,6 +348,174 @@ private void makeMetadataBinary(
     }
   }
 
+  private void makeProjections(
+      final FileSmoosher smoosher,
+      final List<AggregateProjectionSpec> projections,
+      final List<IndexableAdapter> adapters,
+      final IndexSpec indexSpec,
+      final SegmentWriteOutMedium segmentWriteOutMedium,
+      final ProgressIndicator progress,
+      final Closer closer,
+      final Map<String, DimensionMergerV9> parentMergers
+  ) throws IOException
+  {
+    for (AggregateProjectionSpec spec : projections) {
+      final List<IndexableAdapter> projectionAdapters = Lists.newArrayListWithCapacity(adapters.size());
+      for (IndexableAdapter adapter : adapters) {
+        projectionAdapters.add(adapter.getProjectionAdapter(spec.getName()));
+      }
+      // todo (clint): why not from projection spec grouping columns?
+      final List<String> dimensions = projectionAdapters.get(0).getDimensionNames(false);
```

Comment:
Why is the first one special? I see `projectionAdapters.get(0)` referenced a bunch. Is the expectation that they are all the same when it comes to dimensions & column formats?

Reply:
yea, they all have the same schema is the reason, will add some comments to try to make it more obvious